docs/ipdk-dpdk.md

```diff
@@ -0,0 +1,123 @@
+# Building IPDK networking recipe for DPDK target
+
+## Overview
+
+This document describes how to build and run IPDK networking recipe on DPDK target.
+
+## Setup
+
+### Build P4 SDE for DPDK
+```bash
+git clone --recursive git@github.com:p4lang/p4-dpdk-target.git p4sde
+```
+
+For build instructions, refer [P4 SDE Readme](https://github.com/p4lang/p4-dpdk-target/blob/main/README.md#building-and-installing)
+
+### Build and install infrap4d dependencies
+
+```bash
+git clone --recursive git@github.com:ipdk-io/networking-recipe.git ipdk.recipe
+cd ipdk.recipe
+export IPDK_RECIPE=`pwd`
+cd $IPDK_RECIPE/setup
+cmake . -DCMAKE_INSTALL_PREFIX=`path for installing dependencies`
+make
+```
+
+*Note*: If running as non-sudo user, provide `USE_SUDO = ON` option to cmake config and also use `sudo` for ldconfig
+
+### Build Networking Recipe
+
+#### Set environment variables
+ - export DEPS_INSTALL_PATH=`absolute path for installing dependencies`
+ - export SDE_INSTALL=`absolute path for p4 sde install built in previous step`
+ - export LD_LIBRARY_PATH=$IPDK_RECIPE/install/lib/:$SDE_INSTALL/lib:$SDE_INSTALL/lib64:$DEPS_INSTALL_PATH/lib:$DEPS_INSTALL_PATH/lib64:$LD_LIBRARY_PATH
+ 
+
+#### Compile the recipe
+```bash
+cd $IPDK_RECIPE
+./make-all.sh --target=dpdk
+```
+
+### Run Networking recipe
+
+#### Copy the config files to /usr/share/stratum/dpdk required by infrap4d
+
+```bash
+cd $IPDK_RECIPE
+sudo mkdir -p /etc/stratum/
+sudo mkdir -p /var/log/stratum/
+sudo mkdir -p /usr/share/stratum/dpdk
+sudo cp ./install/share/stratum/dpdk/dpdk_port_config.pb.txt /usr/share/stratum/dpdk/
+sudo cp ./install/share/stratum/dpdk/ dpdk_skip_p4.conf /usr/share/stratum/dpdk/
+```
+
+#### Run the infrap4d daemon
+
+```bash
+cd $IPDK_RECIPE
+sudo ./install/sbin/infrap4d
+```
+
+### Run a sample program
+Open a new terminal for setting the pipeline and trying the sample P4 program. Set all the environment variablea and export all environment variables to sudo user
+```bash
+   alias sudo='sudo PATH="$PATH" HOME="$HOME" LD_LIBRARY_PATH="$LD_LIBRARY_PATH"'
```

Comment:
Interesting. I only have to specify `LD_LIBRARY_PATH` to run `infrap4d`.

Reply:
I have tested with this till now. I will test it again just setting LD_LIBRARY_PATH. If not required, will push an update to the document later.